data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -23,19 +33,33 @@
 @DataPrepperPlugin(name = "geoip", pluginType = Processor.class, pluginConfigurationType = GeoIPProcessorConfig.class)
 public class GeoIPProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
 
+  private static final Logger LOG = LoggerFactory.getLogger(GeoIPProcessor.class);
+  private static final String GEO_IP_PROCESSING_MATCH = "geoIpProcessingMatch";
+  private static final String GEO_IP_PROCESSING_MISMATCH = "geoIpProcessingMismatch";
+  private final Counter geoIpProcessingMatchCounter;
+  private final Counter geoIpProcessingMismatchCounter;
+  private final GeoIPProcessorConfig geoIPProcessorConfig;
+  private final String tempPath;
+  private static final String TAGS_ON_MATCH_FAILURE = "tags_on_match_failure";
```

Comment:
It should be "tags_on_geoip_source_not_found". We should not use same tag for all processor. You should also have this in the GeoIP config. The YAML config option can be "tags_on_source_not_found" but the actual tag should have "geoip" in it like I suggested above.

Reply:
Added List<String> tagsOnSourceNotFoundFailure in YAML configuration and necessary changes in respective classes and Readme.MD